vector/src/main/java/im/vector/app/core/utils/DataSource.kt

```diff
@@ -25,6 +25,9 @@ interface DataSource<T> {
 }
 
 interface MutableDataSource<T> : DataSource<T> {
+
+    suspend fun set(value: T)
```

Comment:
I understand where `set` comes from, but it looks weird here for me. Maybe `emit` or `blockingPost` could fit better. What do you think?

Reply:
I don't have very strong feelings towards using a name that's not `set`, but `blockingPost` sounds like something that shouldn't suspend (sort of like blocking the thread its on like `blockingGet`), and `emit` confuses the below `post` methods because ultimately both are `emit` methods

Reply:
well `blockingGet` was just to provide any alternative to `emit` :) I agree that having `emit` and `post` is confusing, but `set` is even more confusing. You don't actually set anything, you just post event and ensure that next event will be posted only after first one was consumed. 

Maybe `postSafely`?

Reply:
`postSafely` assumes that `post` has a chance of throwing which it doesn't. Main reason for opting for `set` is to have those parallels with LiveData's `set` and `post` as the behaviour we ultimately achieve here are the same

Reply:
I'm starting to think `emit` is a good idea know 👀😂

Reply:
and I'm starting to think that param is much better one :) 

@bmarty could you join this conversation? I'd like to hear you thoughts